src/Model/Booking.php

```diff
@@ -850,6 +850,35 @@ public function getEmailSignature(): string {
 		);
 	}
 
+	/**
+	 * Will return if a booking is affected by a total breakdown ( the booked item is not usable ).
+	 * Because since #866 not all total breakdowns are cancelled, we need a way to make sure that the user
+	 * will not be notified about their upcoming bookings or asked to give feedback because they might not have used the item.
+	 * @return bool true if booking is affected by a total breakdown
+	 */
+	public function hasTotalBreakdown(): bool {
+		$itemID = $this->getItem()->ID;
+		$locationID = $this->getLocation()->ID;
+		$restrictions = \CommonsBooking\Repository\Restriction::get(
+			[$locationID],
+			[$itemID],
+			null,
+			true
+		);
+		foreach ( $restrictions as $restriction ) {
+			if (
+				$restriction->getType() == Restriction::TYPE_REPAIR && $restriction->isActive()) {
+				$bookings = \CommonsBooking\Repository\Booking::getByRestriction( $restriction );
```

Comment:
Kann ich nicht hier einfach nur die schauen ob die Item und Location (des Booking Models) mit meiner `$restriction` matchen und dann die Timeranges der beiden Objekte überlappen, anstatt die Datenbank dazu abzufragen und alle Bookings zu iterieren die in einer Restriction liegen?

Reply:
Und letztlich muss er ja wahrscheinlich über keine sonderlich große Zahl von Buchungen iterieren.

Reply:
Oder hast du eine schnelle Idee?

Reply:
Weil du auf "resolved" geklickt hast: Können wir das so lassen?